app/screens/browse_channels/channel_list_row.tsx

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {
+    Text,
+    TouchableOpacity,
+    View,
+} from 'react-native';
+
+import CompassIcon from '@components/compass_icon';
+import {useTheme} from '@context/theme';
+import {makeStyleSheetFromTheme, changeOpacity} from '@utils/theme';
+import {typography} from '@utils/typography';
+
+type Props = {
+    channel: Channel;
+    onPress: (channel: Channel) => void;
+    testID?: string;
+}
+
+export default function ChannelListRow({
+    channel,
+    onPress,
+    testID,
+}: Props) {
+    const theme = useTheme();
+    const style = getStyleFromTheme(theme);
+
+    const handlePress = () => {
```

Comment:
useCallback ?

Reply:
Depends both on channel and onpress, so the only prop change that may cause a re-render and is not affecting this function is testID, which should never change during the lifetime.